compiler/plugins/target/AMD-AIE/aievec/AIEVecOps.td

```diff
@@ -130,39 +129,43 @@ def AIEVec_SRSOp:
 }
 
 def AIEVec_MatMulOp:
-  AIEVec_Op<"matmul", [
-    Pure,
-    AllRanksMatch<["lhs", "rhs", "acc"]>,
-    AllTypesMatch<["acc", "result"]>,
-    ShapesCompatibleWithContraction<"lhs", "rhs", "acc">,
-    IsValidAIE2MatMulShapeAndType<"lhs", "rhs", "acc">
-  ]>,
-  Arguments<(ins AIE2MatMulLHS:$lhs,
-                 AIE2MatMulRHS:$rhs,
-                 AIE2MatMulACC:$acc)>,
-  Results<(outs AIE2MatMulACC:$result)> {
-  let summary = "AIE2 matrix-multiply and accummulate";
+  AIEVec_Op<"matmul", [Pure, AllTypesMatch<["acc", "result"]>]>,
+   Arguments<(ins AnyVectorOfNonZeroRank:$lhs,
+                  AnyVectorOfNonZeroRank:$rhs,
+                  AnyVectorOfNonZeroRank:$acc)>,
+  Results<(outs AnyVectorOfNonZeroRank:$result)> {
+  let summary = "AIE matrix-multiply and accummulate";
   let description = [{
-    AMD AIEv2-specific intrinsic that performs a matrix multiplications
+    AMD AIE-specific intrinsic that performs a matrix multiplications
     between `lhs` and `rhs`, and accumulates the result in `acc`.
 
-    Currently, this intrinsic supports the following type combinations:
-
-         lhs                | rhs                | Accumulator
-        :------------------:|:------------------:|:-----------------:
-         `vector<4x16xi8>`  | `vector<16x8xi4>`  | `vector<4x8xi32>`
-         `vector<4x8xi8>`   | `vector<8x8xi8>`   | `vector<4x8xi32>`
-         `vector<4x4xi16>`  | `vector<4x8xi8>`   | `vector<4x8xi32>`
-         `vector<4x2xi16>`  | `vector<2x8xi16>`  | `vector<4x8xi32>`
-         `vector<2x8xi16>`  | `vector<8x8xi8>`   | `vector<2x8xi64>`
-         `vector<4x8xi16>`  | `vector<8x4xi8>`   | `vector<4x4xi64>`
-         `vector<2x4xi16>`  | `vector<4x8xi16>`  | `vector<2x8xi64>`
-         `vector<4x4xi16>`  | `vector<4x4xi16>`  | `vector<4x4xi64>`
-         `vector<4x2xi32>`  | `vector<2x4xi16>`  | `vector<4x4xi64>`
-         `vector<4x8xbf16>` | `vector<8x4xbf16>` | `vector<4x4xf32>`
+    Currently, this intrinsic supports the following type combinations
```

Comment:
> A few XLLVM ops are removed: we don't use broadcast or a few others, so I've trimmed the set of ops we support down.

Why though? Can't we just keep supporting them?

Reply:
Just to check: are you referring to the reduced set of AIE2 matmuls supported, the removal of ops we don't use (like broadcast) or both? 

The code and ops I've removed is not tested (here or in mlir-aie afaict), not reachable through the compiler currently, and is kind of just a burden to maintain. I don't see any harm in removing it, we can add it back with tests when we need to.